core_helpers/mobile_app_helper.py

```diff
@@ -390,6 +392,73 @@ def scroll_forward(self, distance=150):
             self.exceptions.append("An exception occurred when scrolling forward")
         return result_flag
 
+    def switch_context(self, context_name=None, retries=3):
+        "Switch context in Android app"
+        result_flag = False
+        try:
+            if self.driver.current_context == "WEBVIEW_chrome":
+                time.sleep(1) # Wait on the Chrome tab for a sec before closing, useful for UI debugging
+                for handle in self.driver.window_handles: # Get all open tabs
+                    self.driver.switch_to.window(handle)
+                    # Close the Chrome webview tab,prevents incorrect URL returned in current_url method due to multiple tabs being present
+                    self.driver.close()
+                self.write("Closed current tab on Webview","debug")
+            # Wait to check if the intended context becomes available
+            # This is a fail safe, implemented to address corner case where the context might be avialable after a sec or two
+            for attempt in range(1,retries+1):
+                if not context_name in self.driver.contexts:
+                    self.write(f"Intended context {context_name} not present in available contexts {self.driver.contexts}, retry attempt {attempt}", "debug")
+                    time.sleep(attempt)
+            # Switch context
+            self.driver.switch_to.context(context_name)
+            result_flag = True if self.driver.current_context == context_name else False
+        except NoSuchContextException:
+            self.write(f"No {context_name} context present", 'debug')
```

Comment:
Please set log level to critical for exceptions. (Please do it for only newly added methods. Others, we did it in another PR.)

Reply:
The duplicate methods look to be because of a poor rebase @rohandudam. I have fixed it now.